### DISCUSSION TITLE
Extension not found

### DISCUSSION THREAD

**Discussion by remchuk** [coder/code-server#6358] [Q&A] [answered]
Hey !
So I'm trying to create a podman container which will have vscode and some extensions.

To do this I'm trying to build an image with podman build.

How ever some extensions like "Dev container" or "IntelliCode"
Are always prompts "Failed installing extension, not found"
```
Extension 'visualstudioexptteam.vscodeintellicode' not found.
Make sure you use the full extension ID, including the publisher, e.g.: ms-dotnettools.csharp
Failed Installing Extensions: visualstudioexptteam.vscodeintellicode
```

Now I have read for hours and what I found is that CodeServer Can't use the Microsoft market due to terms of use,

But is there still a way to do it some how? Is there a way around it ? Maybe a different market ?

I have tried this but it didn't seem to work :(
```
# Download and install code-server
RUN curl -fsSL https://code-server.dev/install.sh | sh

RUN export EXTENSIONS_GALLERY='{"serviceUrl": "https://marketplace.visualstudio.com/"}'

RUN code-server --install-extension KevinRose.vsc-python-indent && \
         code-server --install-extension VisualStudioExptTeam.intellicode-api-usage-examples && \
         code-server --install-extension ms-vscode-remote.remote-containers && \
         code-server --install-extension VisualStudioExptTeam.vscodeintellicode
```

**Comment by code-asher** [comment, marked as answer]
The extension appears to be closed-source so I think there is no 
legal avenue for installing on code-server.

I think that is the wrong URL for their gallery but it cannot be 
used legally due to licensing restrictions anyway.  I believe 
manually downloading the vsix from the marketplace and installing 
it from the vsix file is also prohibited.

See https://github.com/microsoft/vscode/issues/31168 for more 
information.
